### ISSUE TITLE
Interface Module Not Working on PfSense 23.01-RELEASE

### ISSUE THREAD

**Issue by bit2pixel** [pfsensible/core#56]
Hi folks,

First time user here. I was able to add VLANs, but unable to add interfaces.
I did a little digging and added a couple more print statements to see the actual error, pasted below.

Thanks!

### Ansible-Playbook Version

```
core 2.14.3
```

### PfSensible Version

```
Latest main (master) branch.
```

### OS and Python Version

```
Python 3.9 (3.11 gives the same result)
MacOS Ventura 13.2.1
```

### PfSense version

```
23.01-RELEASE (amd64) 
built on Fri Feb 10 20:06:33 UTC 2023 
FreeBSD 14.0-CURRENT 
```

### My Playbook

```yaml
- name: Create Interfaces
  hosts: pfsense
  gather_facts: yes
  tasks:
    - name: Read Database CSV
      read_csv:
        path: database.csv
      delegate_to: localhost
      register: db
    - name: Create Interfaces
      loop: "{{ db.list }}"
      loop_control:
        loop_var: item
      pfsensible.core.pfsense_interface:
        interface: "ix1.{{ item.id }}"
        descr: "{{ item.label }}"
        enable: True
        state: present #absent
```

### Log 1

```php
require_once('/etc/inc/interfaces.inc');
$portlist = get_interface_list(); /* add wireless clone interfaces */
if
(is_array($config['wireless']['clone']) && count($config['wireless']['clone']))
	foreach ($config['wireless']['clone'] as $clone)
		$portlist[$clone['cloneif']] = $clone; /* add VLAN interfaces */
if (
	is_array($config['vlans']['vlan']) &&
	count($config['vlans']['vlan'])
) foreach ($config['vlans']['vlan'] as $vlan)
		$portlist[$vlan['vlanif']] = $vlan; /* add
		   Bridge interfaces */
if (is_array($config['bridges']['bridged']) && count($config['bridges']['bridged'])) foreach ($config['bridges']['bridged'] as $bridge)
		$portlist[$bridge['bridgeif']] = $bridge; /* add GIF interfaces */
if
(is_array($config['gifs']['gif']) && count($config['gifs']['gif'])) foreach ($config['gifs']['gif'] as $gif)
		$portlist[$gif['gifif']] = $gif; /* add GRE interfaces */
if (
	is_array($config['gres']['gre']) &&
	count($config['gres']['gre'])
) foreach ($config['gres']['gre'] as $gre)
		$portlist[$gre['greif']] = $gre; /* add LAGG
		   interfaces */
if (is_array($config['laggs']['lagg']) && count($config['laggs']['lagg'])) foreach ($config['laggs']['lagg'] as $lagg) {
		$portlist[$lagg['laggif']] = $lagg; /* LAGG members cannot be assigned */
		$lagifs
			= explode(',', $lagg['members']);
		foreach ($lagifs as $lagif)
			if (isset($portlist[$lagif]))
				unset($portlist[$lagif]);
	} /* add QinQ interfaces */
if (is_array($config['qinqs']['qinqentry']) && count($config['qinqs']['qinqentry'])) foreach ($config['qinqs']['qinqentry'] as $qinq) {
		$portlist["{$qinq['vlanif']}"] = $qinq; /* QinQ members */
		$qinqifs =
			explode(' ', $qinq['members']);
		foreach ($qinqifs as $qinqif)
			$portlist["{$qinq['vlanif']}.{$qinqif}"] = $qinqif;
	} /*
	 add PPP interfaces */
if (is_array($config['ppps']['ppp']) && count($config['ppps']['ppp'])) foreach ($config['ppps']['ppp'] as $pppid => $ppp)
		$portlist[$ppp['if']] = $ppp;
if (is_array($config['openvpn'])) {
	if
	(is_array($config['openvpn']['openvpn-server']))
		foreach ($config['openvpn']['openvpn-server'] as $s)
			$portlist["ovpns{$s['vpnid']}"] = $s;
	if (is_array($config['openvpn']['openvpn-client'])) foreach ($config['openvpn']['openvpn-client'] as $c)
			$portlist["ovpnc{$c['vpnid']}"] = $c;
}
$ipsec_descrs =
	interface_ipsec_vti_list_all(); foreach ($ipsec_descrs as $ifname => $ifdescr)
	$portlist[$ifname] = array(
		'descr' =>
		$ifdescr
	);
echo json_encode($portlist, JSON_PRETTY_PRINT);

----------------------------------------------------

Fatal error: Uncaught TypeError: Cannot access offset of type string on string in Standard input code:2
Stack trace:
    #0 {main}
  thrown in Standard input code on line 2
PHP ERROR: Type: 1, File: Standard input code, Line: 2, Message: Uncaught TypeError: Cannot access offset of type string on string in Standard input code:2
Stack trace:
    #0 {main}
  thrown
```

### Log 2

```python
Traceback (most recent call last):
  File \"/root/.ansible/tmp/ansible-tmp-1678194212.721174-67614-29264918768809/AnsiballZ_pfsense_interface.py\", line 107, in <module>
    _ansiballz_main()
  File \"/root/.ansible/tmp/ansible-tmp-1678194212.721174-67614-29264918768809/AnsiballZ_pfsense_interface.py\", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File \"/root/.ansible/tmp/ansible-tmp-1678194212.721174-67614-29264918768809/AnsiballZ_pfsense_interface.py\", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.pfsensible.core.plugins.modules.pfsense_interface', init_globals=dict(_module_fqn='ansible_collections.pfsensible.core.plugins.modules.pfsense_interface', _modlib_path=modlib_path),
  File \"/usr/local/lib/python3.9/runpy.py\", line 225, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File \"/usr/local/lib/python3.9/runpy.py\", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File \"/usr/local/lib/python3.9/runpy.py\", line 87, in _run_code
    exec(code, run_globals)
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/modules/pfsense_interface.py\", line 159, in <module>
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/modules/pfsense_interface.py\", line 154, in main
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/module_utils/module_base.py\", line 223, in run
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/module_utils/interface.py\", line 177, in _validate_params
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/module_utils/interface.py\", line 385, in _get_interface_list
  File \"/tmp/ansible_pfsensible.core.pfsense_interface_payload_dnr2ahcy/ansible_pfsensible.core.pfsense_interface_payload.zip/ansible_collections/pfsensible/core/plugins/module_utils/pfsense.py\", line 618, in php
  File \"/usr/local/lib/python3.9/json/__init__.py\", line 346, in loads
    return _default_decoder.decode(s)
  File \"/usr/local/lib/python3.9/json/decoder.py\", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File \"/usr/local/lib/python3.9/json/decoder.py\", line 355, in raw_decode
    raise JSONDecodeError(\"Expecting value\", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 2 column 1 (char 1)
```

**Comment by opoplawski** [comment]
I don't have pfSense Plus 23.01 test box at the moment so I'm not sure I'll be able to do much until I do.  Can you run your play with a `-v` option to capture the loop values?  Thanks.

**Comment by opoplawski** [comment]
Managed to setup a pfSense Plus 23.01 VM and reproduce the issue.  Please test out the pfplus23 branch to see if it works for you.  I haven't done much other testing so we may hit other issues with 23.01.

**Comment by bit2pixel** [comment]
Hey @opoplawski thanks a ton for looking into this! I'm currently traveling but I'll check it out as soon as I have access to my server and will let you know.

**Comment by robertkarsai** [comment]
Hi All,

Bumped into the same problem on 2.7.0-DEVELOPMENT, I've tried the interfaces.py from pfplus23 branch and it works now as expected. Thanks for the fix, perhaps it could be merged.

BR
Robert

**Comment by opoplawski** [comment]
This is released now in 0.5.3
